src/charm.py

```diff
@@ -223,6 +244,10 @@ def db_initialised(self, value):
                 f"'db_initialised' must be a boolean value. Proivded: {value} is of type {type(value)}"
             )
 
+    @property
+    def _juju_has_secrets(self) -> bool:
+        return JujuVersion.from_environ().has_secrets
```

Comment:
I think this is another "expensive" juju call each time. 
I used the internal `_jujuversion` instance attribute to make sure not to issue the command every single time we refer to the version of Juju. I'd suggest to put that back.

Reply:
Not sure how this operation is "expensive" to be honest 😄

Reply:
OK, you're right, [apparently](https://ops.readthedocs.io/en/latest/#ops.JujuVersion.from_environ) it's just checking an env var. 

I thought it was a more elaborate call (like the ones behind secret-related operations).